lib/Test/Pod/CoverageChange.pm

```diff
@@ -0,0 +1,235 @@
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+example: ['lib', 'other directory'] | 'lib'
+
+=item C<$allowed_naked_packages> These packages are allowed to have naked subs equal to specified numbers. (optional)
+
+example: {Package1 => 2, Package2 => 1, Package3 => 10}
+
+=item C<$ignored_packages> - A list of packages that will be ignored in our checks, supports arrayref. (optional)
+
+example: ['MyPackage1', 'MyPackage2', 'MyPackage3']
+
+=back
+
+=cut
+
+sub pod_coverage_syntax_ok {
+    my $path = shift;
+    my $allowed_naked_packages = shift // {};
+    my $ignored_packages = shift // [];
+
+    $path = [ $path ] unless ref $path eq 'ARRAY';
+    $ignored_packages = [ $ignored_packages ] unless ref $ignored_packages eq 'ARRAY';
+
+    check_pod_coverage($path, $allowed_naked_packages, $ignored_packages);
+    check_pod_syntax($path, $ignored_packages);
+}
+
+=head2 check_pod_coverage
+
+Checks POD coverage for all the modules that exists under a given directory.
+Passes the $allowed_naked_packages to the L<Test::Pod::CoverageChange::check_allowed_naked_packages>.
+Ignores the packages in the C<$ignored_packages> parameter.
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item C<$allowed_naked_packages> These packages are allowed to have naked subs equal to specified numbers. (optional)
+
+=item C<$ignored_packages> - A list of packages that will be ignored in our checks, supports arrayref. (optional)
+
+=back
+
+=cut
+
+sub check_pod_coverage {
+    my $path = shift;
+    my $allowed_naked_packages = shift;
+    my $ignored_packages = shift;
+
+    check_allowed_naked_packages($allowed_naked_packages, $ignored_packages) if keys %$allowed_naked_packages;
+
+    # Check for newly added packages PODs
+    my @ignored_packages = (keys %$allowed_naked_packages, @$ignored_packages);
+    foreach my $package (Test::Pod::Coverage::all_modules(@$path)) {
+        next if @ignored_packages && (any {$_ eq $package} @ignored_packages);
+        pod_coverage_ok($package, { private => [] });
+    }
+}
+
+=head2 check_pod_syntax
+
+Check POD syntax for all the modules that exists under a given directory.
+
+=over 4
+
+=item * C<$path> - path or arrayref of directories to check (recursively)
+
+=item C<$ignored_packages> - A list of packages that will be ignored in the checks, supports arrayref. (optional)
+
+=back
+
+=cut
+
+sub check_pod_syntax {
+    my $path = shift;
+    my $ignored_packages = shift;
+    my $Test_Builder = Test::More->builder;
+
+    my @ignored_packages_full_path = ();
+    for (@$ignored_packages) {
+        my $file_path = module_path($_);
+        push(@ignored_packages_full_path, $file_path) if defined $file_path;
+    }
+
+    my @files_path = File::Find::Rule->file()
+        ->name('*.p[m|l]')
+        ->in(@$path);
+
+    for my $file_path (@files_path) {
+        chomp $file_path;
```

Comment:
why do we need this?

Reply:
I put this there to make sure that no extra space/line exists in the $file_path variable. It can cause the grep function to do an unexpected matching.

Reply:
but why would that happen? do you have cases where File::Find::Rule is returning entries with a linefeed?

Reply:
No, I faced this issue while I was using another method to get the files' name list. https://github.com/binary-com/perl-Test-Pod-CoverageChange/pull/2/commits/1227d8a8ef945bb382a72d77082d3d19d3b24cae#diff-7bf492e62d5c3c029718bd64d48095dbR55.